adaptive/tests/test_average_learner1d.py

```diff
@@ -0,0 +1,48 @@
+import numpy as np
+import pandas as pd
+from pandas.testing import assert_series_equal
+
+from adaptive import AverageLearner1D
+from adaptive.tests.test_learners import (
+    generate_random_parametrization,
+    noisy_peak,
+    simple_run,
+)
+
+
+def almost_equal_dicts(a, b):
+    assert_series_equal(pd.Series(sorted(a.items())), pd.Series(sorted(b.items())))
+
+
+def test_tell_many_at_point():
+    f = generate_random_parametrization(noisy_peak)
+    learner = AverageLearner1D(f, bounds=[-2, 2])
+    control = AverageLearner1D(f, bounds=[-2, 2])
+    learner._recompute_losses_factor = 1
+    control._recompute_losses_factor = 1
+    simple_run(learner, 100)
+    for x, samples in learner._data_samples.items():
+        control.tell_many_at_point(x, samples)
+
+    almost_equal_dicts(learner.data, control.data)
+    almost_equal_dicts(learner.error, control.error)
+    almost_equal_dicts(learner.rescaled_error, control.rescaled_error)
+    almost_equal_dicts(learner.neighbors, control.neighbors)
+    almost_equal_dicts(learner.neighbors_combined, control.neighbors_combined)
+    assert learner.npoints == control.npoints
+    assert learner.nsamples == control.nsamples
+    assert len(learner._data_samples) == len(control._data_samples)
+    assert learner._data_samples.keys() == control._data_samples.keys()
+
+    for k, v1 in learner._data_samples.items():
+        v2 = control._data_samples[k]
+        assert len(v1) == len(v2)
+        np.testing.assert_almost_equal(np.sort(v1), np.sort(v2))
+
+    assert learner._bbox[0] == control._bbox[0]
+    assert learner._bbox[1] == control._bbox[1]
```

Comment:
@AlvaroGI, @akhmerov, by adding this test I found a potential bug in the algorithm because it fails at this point.

I believe that the correct behavior for setting the bounding box in "y" is that it's set by the min and max average of the data.
This correctly happens in `tell_many_at_point` in `super()._update_scale(x, y_avg)`, however, when adding points one by one using `learner.tell` (starting with points far away from the mean) the bounding box will be defined by these points far away from the mean.

We can solve this by defining the bounding box:
- by its extremes `min([min(x) for x in learner._data_samples.values()]), max([max(x) for x in learner._data_samples.values()])` always (simplest solution but perhaps less correct)
- define the bounding box by the means of the data at each point, this means a lot of bookkeeping and perhaps (a lot?) extra computions

Reply:
Hi Bas,
Sorry for the delay in my answer, I needed to take some time to think about this.

__Some findings after looking at this issue:__
 1. When the samples are added by running the learner, the maximum and minimum values of `learner._bbox` indeed correspond to the maximum and minimum values of `y` over all samples ("undesired" behavior), as you said. This is because the old `learner._bbox` values are compared to each new sample.
 2. However, when using `tell_many_at_point()`, I found that the `learner._bbox` values do *not* necessarily correspond to the maximum and minimum values of the data (data = average over all samples at each point).  In this case, old `learner._bbox` values are compared to the average of all the many samples that are added through `tell_many_at_point()`. Therefore, if you add points to a new learner, this corresponds to the maximum and minimum value of the data ("desired" behavior); but if we run a learner for a while and then add new samples, the values of `learner._bbox` will be compared to the mean value of the new samples, and we can end up with a `learner._bbox` that does not correspond to the max/min of the data nor to the max/min of the samples.
 3. Another bug I found in `tell_many_at_point()` is that the new value of the data at each point adopted the value of the mean of the new data samples, instead of the mean over all data samples (new and old) at that same point. I have already fixed this in my last commit.

I agree that the correct behavior should be the one you explained. However, I don't think `learner._bbox` has a dramatic impact on the performance of the algorithm (it is only used to compute `learner._scale`, which is mainly used to provide a finite loss to intervals with infinite loss, is that right?), so any of both behaviors could be fine in practice.
Therefore, I would say we should go for your first solution, which I think requires far less computations. To do that, I think we just need to bugfix `tell_many_at_point()` by replacing line 397 in `average_learner1D.py`:
```
super()._update_scale(x, self.data[x])
```
by the following two lines:
```
super()._update_scale(x, min(self._data_samples[x])
super()._update_scale(x, max(self._data_samples[x])
```

What do you think?